packages/wdio-junit-reporter/src/index.ts

```diff
@@ -264,7 +264,7 @@ class JunitReporter extends WDIOReporter {
             const suite = this.suites[suiteKey]
             if (isCucumberFrameworkRunner && suite.type === type && specFileName === suite.file) {
                 builder = this._addCucumberFeatureToBuilder(builder, runner, specFileName, suite)
-            } else if (!isCucumberFrameworkRunner) {
+            } else if (!isCucumberFrameworkRunner && specFileName === suite.file) {
```

Comment:
This implementation is also different, is that intentional?

Reply:
it is not different. it is looking different because on v8 we have some changes, and v7 has all the things like this.
v7 & v8 implementations are doing the same thing.